code/game/machinery/excercisebike.dm

```diff
@@ -0,0 +1,57 @@
+/obj/machinery/power/excercise_bike
+	name = "Excercise bike"
+	desc = "The new NT 4FA5 AS5 excercise bike allows you to power the station whilst excercising: It's fitted with an efficient mechanism for power generation."
+	icon = 'icons/obj/machines/excercise_bike.dmi'
+	icon_state = "bike-off"
+	can_buckle = TRUE
+	max_buckled_mobs = 1
+	var/power = 200 //How many watts are generated per pedal turn
+	var/delay = 12 //Delay to avoid ultraspam
+	var/cooldown_time = 0
+	anchored = TRUE
+	can_be_unanchored = TRUE
+	buckle_lying = 0
+	circuit = /obj/item/circuitboard/machine/excercise_bike
+
+/obj/machinery/power/excercise_bike/examine(mob/user)
+	. = ..()
+	if(powernet)
+		to_chat(user, "<span_class='warning'>It's connected to a powernet</span>")
+	else
+		to_chat(user, "<span_class='warning'>It's not connected to a powernet, and will not generate power.</span>")
+
+/obj/machinery/power/excercise_bike/attackby(obj/item/W, mob/user)
+	if(default_unfasten_wrench(user, W, time = 20))
+		powernet = null
+		return
+	return ..()
+
+/obj/machinery/power/excercise_bike/RefreshParts()
+	power = 0
+	for(var/obj/item/stock_parts/capacitor/C in component_parts)
```

Comment:
are cable coils a part of component_parts? If not you can make this typeless.

Probably is tho :(

Reply:
Yeah cables are in component parts :(